varinfo/var_info.py

```diff
@@ -189,6 +189,38 @@ def get_variables_with_coordinates(self) -> Dict[str, OutputVariableType]:
                 if variable.references.get('coordinates') is not None
                 and not self.variable_is_excluded(variable, exclusion_pattern)}
 
+    def _is_spatial_temporal_dimension(self, dimension_path: str) -> bool:
+        """ A helper method simplifying the list comprehension in
+            `is_science_variable`.
+        """
+        dimension = self.get_variable(dimension_path)
+        return (dimension is not None
+                and (dimension.is_geographic() or dimension.is_temporal()
+                     or dimension.is_projection_x_or_y()))
+
+    def is_science_variable(self, variable: OutputVariableType) -> bool:
+        """ Determine if a variable is a science variable.
+            A science variable is classified as a variable that is NOT
+            a spatial temporal dimension variable. But contains
+            geographic, temporal, and or projected spatial dimensions, or
+            a coordinate or grid mapping reference attribute.
+
+        """
+        if any(
+            self._is_spatial_temporal_dimension(dimension)
+            for dimension in variable.dimensions
+            if dimension != variable.full_name_path
+            and not variable.full_name_path.endswith('_bnds')
+        ):
+            return True
+
+        if (
+            variable.references.get('coordinates') is not None
+                or variable.references.get('grid_mapping')) is not None:
+            return True
+
```

Comment:
That is almost certainly not what you intended if you are trying to make sure that neither `coordinates` nor `grid_mapping`  are None.   

If this works, I think this works by accident, you probably want to move the parenthesis for readability.